### PR TITLE
MDCT-1993 api updates

### PR DESCRIPTION
**Pull request by gmrabian**
## Description
[MDCT-1993](https://qmacbis.atlassian.net/browse/MDCT-1993) modify api to match standards
1. Fetch to a specific item returns a 404 when the response is empty (instead of 200)
2. Create calls return a 201 instead of 200
3. Fetch’s with full primary keys will convert to a get from a query
4. An update where the matching record does not exist returns a 404 (instead of a 500)

### How to test
Try out the [mcr postman collection from confluence](https://qmacbis.atlassian.net/l/cp/Me1mwh6p)! Check that:
- Create now returns a 201 instead of 200
- Update without existing item returns 404
- Update without new data returns 400
- Fetch where specific item doesn't exist returns 404 (does not apply to queries like "reportsByState")

ping me if you want help!

### Changed Dependencies
none

## Code author checklist
- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://bit.ly/3zPrxuZ) tests
- [ ] I have added analytics, if necessary
- [ ] I have updated the documentation, if necessary

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

File: services/app-api/handlers/banners/fetch.test.ts [hunk 1]
```diff
@@ -3,21 +3,19 @@ import { APIGatewayProxyEvent } from "aws-lambda";
...
-      },
-    }),
+    get: jest
+      .fn()
+      .mockImplementationOnce(() => ({
+        Item: undefined,
+      }))
+      .mockImplementation(() => ({
+        Item: mockBannerResponse,
+      })),
```
Comment (gmrabian) [inline services/app-api/handlers/banners/fetch.test.ts:18]: Note: I'm planning on opening a stacked branch to improve this

Comment (gmrabian) [inline services/app-api/handlers/banners/fetch.test.ts:18]: Note: I'm planning on opening a stacked branch to improve this

Comment (gmrabian) [reply services/app-api/handlers/banners/fetch.test.ts:18]: Stacked PR here: https://github.com/CMSgov/mdct-mcr/pull/4181

**Review by karla-vm — APPROVED** [review]
_nice_ ✨  I was able to use the Postman Collection too!

**Review by scwambach — APPROVED** [review]

**Pull request merged**